R/RcppExports.R

```
@@ -145,3 +145,11 @@ log_level <- function(level) {
     .Call('_httpuv_log_level', PACKAGE = 'httpuv', level)
 }
 
+#' Get HTTP status description
+#'
+#' @keywords internal
+#' @export
+getStatusDescription <- function(code) {
+    .Call('_httpuv_getStatusDescription', PACKAGE = 'httpuv', code)
+}
```

This function should gracefully handle both reasonably and completely unexpected user input.

```r
httpuv::getStatusDescription("404")
#> Error: Not compatible with requested type: [type=character; target=integer].
```

If the input must be numeric, it'd be better to throw an error saying so. Alternatively, if you think it's worth the effort, something like the above could be made to work.

I can also see an argument for `getStatusDescription()` to never throw and to instead return `NA` if no description is found.

---

Its just to avoid replicating code, but I get what you are saying. RestRServe has https://github.com/rexyai/RestRserve/blob/2f84b697d1d0827340234b25167e75f519b1a013/R/http_status_codes.R. plumber could do the same I guess. Thanks for taking the time to respond.

---

If the target is plumber, IMHO I'd rather plumber maintain its own, more complete list, which would make it easier to build user-facing helper functions, for example to choose the correct status code for a response, etc. I think it'd be quite okay to start by copying the code list from httr2 to seed the initial list.